onnxruntime/core/providers/cpu/tensor/onehot.cc

```diff
@@ -152,8 +169,22 @@ Status OneHotOp<in_type, out_type, depth_type>::Compute(OpKernelContext* p_op_ke
 
   // Split indices into matrix of size prefix_dim_size x suffix_dim_size
   Eigen::array<Eigen::DenseIndex, 2> indices_dims_e = {
-    {static_cast<Eigen::DenseIndex>(prefix_dim_size), static_cast<Eigen::DenseIndex>(suffix_dim_size)}};
+      {static_cast<Eigen::DenseIndex>(prefix_dim_size), static_cast<Eigen::DenseIndex>(suffix_dim_size)}};
+
+  // Handle negative indices. It's faster to create a new indices instead of comparing in generator
+  // since generator has much larger loops.
   const auto* indices_data = indices->Data<in_type>();
+  std::vector<in_type> adjusted_indices;
+  adjusted_indices.reserve(indices_shape.Size());
```

Comment:
nit: `indices_shape.Size()` can possibly be cached rather than call Size() twice in consecutive lines.